SETUP.md

```diff
@@ -38,51 +61,95 @@ nvm alias default 8
 ```
 
-Re-login to the system and run:
+Optionally, install Jenv to manage multiple Java versions
 
 ```bash
-rustup install 1.36.0
-rustup default 1.36.0
+brew install jenv
+eval "$(jenv init -)"
+jenv add /Library/Java/JavaVirtualMachines/<java8 version here>/Contents/Home
 ```
 
-If you're building Geth for Android, you require an NDK that has a cross-compilation toolchain. You can get it by appropriately defining the relevant environment variables, e.g.:
+### Linux
+
+#### Install Node, Yarn and friends
+
+We use Yarn to build all of the [celo-monorepo] repo.
+
+Install `nvm` (allows you to manage multiple versions of Node), Node 8 and `yarn`:
 
 ```bash
-export NDK_VERSION=android-ndk-r19c
-export ANDROID_NDK=ndk_bundle/android-ndk-r19c
+# Installing Node
+sudo apt-get update
+sudo apt-get install nodejs
+
+# Installing Nvm
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | bash
+source ~/.bashrc
+
+# Setting up the right version of Nvm
+nvm install 8
+nvm alias default 8
+
+# Installing Yarn - https://yarnpkg.com/en/docs/install#debian-stable
+curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
+echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+sudo apt-get update && sudo apt-get install yarn
 ```
 
-and running `make ndk_bundle`. This will download the NDK for your platform.
+#### Installing OpenJDK 8
 
-### Java
+```
+sudo apt install openjdk-8-jdk
+```
 
-We need Java to be able to build and run Android to deploy the mobile app to
-test devices. Android currently only builds correctly with Java 8. (Using
-OpenJDK because of [Oracle being Oracle][oracle being oracle])
+### Some common stuff
+
+
+#### Optional: Install Rust
+
+We use Rust to build the [bls-zexe](https://github.com/celo-org/bls-zexe) repo, which Geth depends on. If you only use the monorepo, you probably don't need this.
 
 ```bash
-brew install cask
-brew tap homebrew/cask-versions
-brew cask install homebrew/cask-versions/adoptopenjdk8
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
 ```
 
-Optionally, install Jenv to manage multiple Java versions
+Now lets add Rust to the PATH:
+```
+echo "export PATH=$PATH:~/.cargo/bin/" >> ~/.bashrc
+source ~/.bashrc
+```
+
+With Rust binaries in your PATH you should be able to run:
 
 ```bash
-brew install jenv
-eval "$(jenv init -)"
-jenv add /Library/Java/JavaVirtualMachines/<java8 version here>/Contents/Home
+rustup install 1.36.0
+rustup default 1.36.0
+```
+
+If you're building Geth for Android, you require an NDK that has a cross-compilation toolchain. You can get it by appropriately defining the relevant environment variables, e.g.:
+
+```bash
+export NDK_VERSION=android-ndk-r19c
+export ANDROID_NDK=ndk_bundle/android-ndk-r19c
 ```
 
-### Install Android Dev Tools
+and running `make ndk_bundle`. This will download the NDK for your platform.
+
+
+#### Install Android Dev Tools
```

Comment:
android-sdk and android-platform-tools still use `brew` here, could be moved to the corresponding OS specific setup along with an Ubuntu version